src/hooks/useErrorToast/useErrorToast.tsx

```diff
@@ -1,20 +1,30 @@
 import { useToast } from '@chakra-ui/react'
+import { ChainAdapterError } from '@shapeshiftoss/chain-adapters'
 import { useCallback } from 'react'
 import { useTranslate } from 'react-polyglot'
+import { InlineCopyButton } from 'components/InlineCopyButton'
+import { RawText } from 'components/Text'
 
 export const useErrorHandler = () => {
   const toast = useToast()
   const translate = useTranslate()
 
   const showErrorToast = useCallback(
     (error: unknown) => {
-      const description = translate('common.generalError')
+      const description =
+        error instanceof ChainAdapterError
+          ? translate(error.metadata.translation, error.metadata.options)
+          : translate('common.generalError')
```

Comment:
Uh oh. I've made a similar change in #8203, but mine uses explicit args rather than a translation string inside the error.

I love the copy button you've added though, so perhaps we can update this to use a function signature closer to mine and keep the copy button?

https://github.com/shapeshift/web/pull/8203/files#diff-ee926a4c2f1f80ce9afd360dc300601dc1b61828c15b1665dc687b762993b5c2 

<img src="https://github.com/user-attachments/assets/28614dd4-12dc-4859-8415-843aa809e2eb" width="200">

Reply:
I see this is going to handle errors way more transparently than explicit args which is a good thing, so I'm happy for you to get this in unchanged and I should be able to adapt my solution to this without borking things.